variables.tf

```diff
@@ -2,8 +2,150 @@
 # Input Variables
 ########################################################################################################################
 
-#variable "my_variable" {
-#  type        = string
-#  description = "A description of my variable"
-#  default     = "default_value"
-#}
+variable "resource_group_id" {
+  description = "The resource group ID where the watsonx services will be provisioned. Required when creating a new instance."
+  type        = string
+  default     = null
+  validation {
+    condition     = var.existing_watson_studio_instance_crn == null ? length(var.resource_group_id) > 0 : true
+    error_message = "You must specify a value for 'resource_group_id', required to create watson studio instance."
+  }
+
+  validation {
+    condition     = var.existing_machine_learning_instance_crn == null ? length(var.resource_group_id) > 0 : true
+    error_message = "You must specify a value for 'resource_group_id', required to create watson machine learning instance."
+  }
+}
+
+variable "prefix" {
+  description = "The name to be used on all Watson resources as a prefix."
+  type        = string
+
+  validation {
+    condition     = var.prefix != "" && length(var.prefix) <= 25
+    error_message = "You must provide a value for the resource_prefix variable and the resource_prefix length can't exceed 25 characters."
```

Comment:
Do we have this prefix length restriction ? 
Moreover the error message mentions a different variable `resource_prefix`.

We do have another validation around prefix i.e. : 
```tf
  validation {
    error_message = "Prefix must begin and end with a letter and contain only letters, numbers, and - characters."
    condition     = can(regex("^([A-z]|[a-z][-a-z0-9]*[a-z0-9])$", var.prefix))
  }
```

Reply:
It is because name is used as - "${var.prefix}-watson-studio" and "${var.prefix}-watson-machine-learning"
I added new variables for service name now hence this restriction can be lifted.